Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -412,6 +412,97 @@ public override int GetHashCode()
             }
         }
 
+        /// <summary>
+        /// Helper class for managing the visibility of the gaze pointer to match windows mixed reality and HoloLens 2
+        /// When application starts, gaze pointer is visible. Then when articulate hands / motion controllers
+        /// appear, hide the gaze cursor. Whenever user says the voice wake word, make the 
+        /// gaze controller appear.
+        /// </summary>
+        private class GazePointerStateMachine : IMixedRealitySpeechHandler
+        {
+            private enum GazePointerState
+            {
+                Initial, // When the application starts up, the gaze pointer should be active
+                GazePointerActive, // Gaze pointer is active when no hands are visible, after "select" 
+                GazePointerInactive // Gaze pointer is inactive as soon as motion controller or articulated hand pointers appear
+            }
+            private GazePointerState gazePointerState = GazePointerState.Initial;
+            private bool activateGazeKeywordIsSet = false;
+
+            public bool IsGazePointerActive
+            {
+                get { return gazePointerState != GazePointerState.GazePointerInactive; }
+            }
+
+            public void UpdateState(HashSet<PointerData> pointers, GenericPointer currentGazePointer)
```

Comment:
This whole loop has a ton of complexity just to gauge the state of a cursor. I understand why it's happening, but I think it should be heavily profiled before being merged in.